pkg/webhook/cronfederatedhpa/validating_test.go

```diff
@@ -0,0 +1,344 @@
+					APIVersion: autoscalingv1alpha1.GroupVersion.String(),
+					Kind:       "Test",
+					Name:       "foo",
+				},
+				Rules: []autoscalingv1alpha1.CronFederatedHPARule{
+					{
+						Name:           "bar",
+						Schedule:       "0 0 13 * 5",
+						TargetReplicas: pointer.Int32(1),
+					},
+				},
+			},
+			expectedErr: fmt.Sprintf("invalid scaleTargetRef kind: %s, only support %s", "Test", autoscalingv1alpha1.FederatedHPAKind),
+		},
+		{
+			name: "unspecified spec.scaleTargetRef.kind",
+			spec: &autoscalingv1alpha1.CronFederatedHPASpec{
+				ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
+					Name: "foo",
+				},
+				Rules: []autoscalingv1alpha1.CronFederatedHPARule{
+					{
+						Name:           "bar",
+						Schedule:       "0 0 13 * 5",
+						TargetReplicas: pointer.Int32(1),
+					},
+				},
+			},
+			expectedErr: "spec.scaleTargetRef.kind: Required value",
+		},
+		{
+			name: "invalid spec.scaleTargetRef.kind: .",
+			spec: &autoscalingv1alpha1.CronFederatedHPASpec{
+				ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
+					Kind: ".",
+					Name: "foo",
+				},
+				Rules: []autoscalingv1alpha1.CronFederatedHPARule{
+					{
+						Name:           "bar",
+						Schedule:       "0 0 13 * 5",
+						TargetReplicas: pointer.Int32(1),
+					},
+				},
+			},
+			expectedErr: "may not be '.'",
+		},
+		{
+			name: "invalid spec.scaleTargetRef.kind: contains /",
+			spec: &autoscalingv1alpha1.CronFederatedHPASpec{
+				ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
+					Kind: "A/B",
+					Name: "foo",
+				},
+				Rules: []autoscalingv1alpha1.CronFederatedHPARule{
+					{
+						Name:           "bar",
+						Schedule:       "0 0 13 * 5",
+						TargetReplicas: pointer.Int32(1),
+					},
+				},
+			},
+			expectedErr: "may not contain '/'",
+		},
+		{
+			name: "unspecified spec.scaleTargetRef.name",
+			spec: &autoscalingv1alpha1.CronFederatedHPASpec{
+				ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
+					Kind: "foo",
+				},
+				Rules: []autoscalingv1alpha1.CronFederatedHPARule{
+					{
+						Name:           "bar",
+						Schedule:       "0 0 13 * 5",
+						TargetReplicas: pointer.Int32(1),
+					},
+				},
+			},
+			expectedErr: "spec.scaleTargetRef.name: Required value",
+		},
+		{
+			name: "invalid spec.scaleTargetRef.name: ..",
+			spec: &autoscalingv1alpha1.CronFederatedHPASpec{
+				ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
+					Kind: "foo",
+					Name: "..",
+				},
+				Rules: []autoscalingv1alpha1.CronFederatedHPARule{
+					{
+						Name:           "bar",
+						Schedule:       "0 0 13 * 5",
+						TargetReplicas: pointer.Int32(1),
+					},
+				},
+			},
+			expectedErr: "may not be '..'",
+		},
+		{
+			name: "invalid spec.scaleTargetRef.name: contains %",
+			spec: &autoscalingv1alpha1.CronFederatedHPASpec{
+				ScaleTargetRef: autoscalingv2.CrossVersionObjectReference{
+					Kind: "foo",
+					Name: "1%100",
```

Comment:
Do we need to test the scaleTargetRef in such detail? Will it make the unit test code difficult to maintain? Essentially, this is a function call to K8s.